### PR TITLE
Add End-to-end test

### LINKED ISSUE #1264: Add End-to-End Test
**Describe the Issue**

Right now, we have unit and instrumentation tests. A nice addition would be an end-to-end (e2e) that launches the app and verifies the app functionality. An example of this would be verifying that the number of resources downloaded on the app is equal to the number of resources that are on the server

### PR DESCRIPTION
**Pull request by omarismail94**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1264 

**Description**
End-to-end test that verifies the number of resources downloaded on the app is equals the number of resources that are on the server for the search query (`address-city=NAIROBI`) provided.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?
No

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | **Code health** | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by jingtang10** [comment]
@omarismail94 thanks so much for this I'm seeing:

```
/data/data/com.google.android.fhir.demo/databases/resources.db: 1 file pulled, 0 skipped. 93.6 MB/s (954368 bytes in 0.010s)
Number of patients in App's DB is: 224
Number of patients in HAPI server is: 224.
```

does this mean this is successful?

**Comment by omarismail94** [comment]
> @omarismail94 thanks so much for this I'm seeing:
> 
> ```
> /data/data/com.google.android.fhir.demo/databases/resources.db: 1 file pulled, 0 skipped. 93.6 MB/s (954368 bytes in 0.010s)
> Number of patients in App's DB is: 224
> Number of patients in HAPI server is: 224.
> ```
> 
> does this mean this is successful?

Yes, I added an extra line at the end to show the test was a success

**Comment by jingtang10** [comment]
> > @omarismail94 thanks so much for this I'm seeing:
> > ```
> > /data/data/com.google.android.fhir.demo/databases/resources.db: 1 file pulled, 0 skipped. 93.6 MB/s (954368 bytes in 0.010s)
> > Number of patients in App's DB is: 224
> > Number of patients in HAPI server is: 224.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > does this mean this is successful?
> 
> Yes, I added an extra line at the end to show the test was a success to make it clear

this is fantastic - thanks so much for adding this.

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: e2e/e2e.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  get_logs_from_device
+  echo "App sync hasn't succeeded. Waiting again..."
+  adb -s emulator-5554 shell am start -n "com.google.android.fhir.demo/.MainActivity"
+done
+
+pull_db
+patient_count_in_app=$(sqlite3 resources.db "select count(*) from ResourceEntity where resourceType = 'Patient';")
+echo "Number of patients in App's DB is: ${patient_count_in_app}"
+patient_count_in_hapi=$(curl -s "http://hapi.fhir.org/baseR4/Patient?address-city=NAIROBI&_summary=count" | jq .total)
+echo "Number of patients in HAPI server is: ${patient_count_in_hapi}"
```
[the remark this answers is not in the document]
Comment (omarismail94) [reply e2e/e2e.sh:87]: I can add a Docker image with a local HAPI FHIR Server!

**Comment by jingtang10** [comment]
closing this for now @omarismail94

**Pull request closed without merging**